docs/l3/api_flow.md

```diff
@@ -0,0 +1,413 @@
+# Prerequisites (dscp-api-identity-service)
+
+## Setting node aliases 
+
+The values set for each persona are your choice, they should provide a recognisable value in response bodies.
+
+Here we have went with:
+
+- “Member A” to represent an order
+
+- “Member B” to represent a capacity
+
+- “Optimiser” to represent an optimiser
+
+1. For each node, set node address `5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY` to “Member A” i.e. Order, using `PUT /members/{address}`
+
+2. For each node, set node address `5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty` to “Member B” i.e. Capacity, using `PUT /members/{address}`
+
+3. For each node, set node address `5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y` to “Optimiser” using `PUT /members/{address}`
+
+## Retrieving node aliases
+
+Using `GET /members` on each node returns a list of all node aliases as set above:
+```
+[
+  {
+    "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
+    "alias": "Member B"
+  },
+  {
+    "address": "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y",
+    "alias": "Optimiser"
+  },
+  {
+    "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
+    "alias": "Member A"
+  }
+]
+```
+# Match Creation API Flow (dscp-matchmaker-api)
+
+## Context
+
+In the case of a logistics matching service where one provider has an order to be moved and another has some capacity to move orders, one might represent an order as a `demand_a` and a capacity as a `demand_b`.
+
+## End-to-End Flow
+
+Parameter files uploaded in `POST /v1/attachment` as Member A & Member B, respectively.
+
+#### Member A
+```
+{
+    "on_pallet": true,
+    "pallet_size": {
+      "max_length": 120,
+      "max_height": 180,
+      "max_width": 100
+    },
+    "item_dimensions": {
+      "length": 80,
+      "height": 50,
+      "width": 40
+    },
+    "item_weight": 25,
+    "pickup_location": {
+      "geo_location": {
+        "latitude": 51.5074,
+        "longitude": -0.1278
+      },
+      "timed_window": {
+        "start_time": "2023-05-10T10:00:00Z",
+        "end_time": "2023-05-10T12:00:00Z"
+      },
+      "expected_time_for_loading": 30
+    },
+    "dropoff_location": {
+      "geo_location": {
+        "latitude": 52.5200,
+        "longitude": 13.4050
+      },
+      "estimated_time": "2023-05-11T15:30:00Z"
+    },
+    "mixing_exclusion_criteria": {
+      "critical": [
+        "toxic",
+        "flammable"
+      ],
+      "user_defined": [
+        "organic",
+        "inorganic"
+      ]
+    },
+    "specialist_requirements": [
+      "refrigerated",
+      "hazardous"
+    ]
+  }
+```
+#### Member B
+```
+
+```
```

Comment:
missing an example parameters file for MemberB?

Reply:
```suggestion
```
{
    "palletised_dimensions": true,
    "pallet_size": {
      "max_length": 120,
      "max_height": 180,
      "max_width": 100
    },
    "max_weight": 500,
    "wheel_position": "front",
    "pickup_locations": [
      {
        "geo_location": {
          "latitude": 51.5074,
          "longitude": -0.1278
        },
        "estimated_time": "2023-05-10T10:00:00Z"
      },
      {
        "geo_location": {
          "latitude": 52.5200,
          "longitude": 13.4050
        },
        "estimated_time": "2023-05-11T15:30:00Z"
      }
    ],
    "required_buffer_time": 30,
    "driver_breaks": {
      "interval": 240,
      "duration": 30
    },
    "latest_end_time": "2023-05-12T18:00:00Z",
    "latest_end_time_buffer": 60,
    "worked_time_so_far": 300,
    "breaks_so_far": 2,
    "vehicle_type": "van",
    "vehicle_specialism": "refrigerated",
    "existing_item_exception_criteria": [
      {
        "name": "fragile",
        "weight_threshold": 10
      },
      {
        "name": "flammable",
        "weight_threshold": 5
      }
    ],
    "distance_travelled_so_far": 150,
    "current_vehicle_location": {
      "latitude": 51.5074,
      "longitude": -0.1278
    },
    "priority_value": 5,
    "timeout_cutoff": 120
  }
```
```